File: packages/main-api/src/types/space_status.rs
use bdk::prelude::*;

#[derive(
    Debug,
    Clone,
    serde_with::SerializeDisplay,
    serde_with::DeserializeFromStr,
    Default,
    DynamoEnum,
    JsonSchema,
    Eq,
    PartialEq,
)]
pub enum SpaceStatus {
    #[default]
    Waiting, // Draft
    InProgress, // Published
    Started,    // Started
    Finished,   // Finished
}

File: flutter-packages/mobile/lib/services/space/types/space_status.dart
enum SpaceStatus { waiting, inProgress, started, finished }

SpaceStatus spaceStatusFromJson(dynamic v) {
  if (v == null) return SpaceStatus.waiting;

  final s = v.toString().toLowerCase();

  switch (s) {
    case 'waiting':
      return SpaceStatus.waiting;
    case 'in_progress':
    case 'inprogress':
      return SpaceStatus.inProgress;
    case 'started':
      return SpaceStatus.started;
    case 'finished':
      return SpaceStatus.finished;
    default:
      return SpaceStatus.waiting;
  }
}

String spaceStatusToJson(SpaceStatus s) {
  switch (s) {
    case SpaceStatus.waiting:
      return 'waiting';
    case SpaceStatus.inProgress:
      return 'in_progress';
    case SpaceStatus.started:
      return 'started';
    case SpaceStatus.finished:
      return 'finished';
  }
}
